book/src/solana-scripts.md

```diff
@@ -0,0 +1,108 @@
+# Solana Scripts
+
+This architecture describes how the Pipeline runtime supports scripts. 
+
+In Solana programs have absolute control over the state assigned to them.  The
+goal of this architecture is provide a simple way for on-chain programs to
+interact with other programs without violating the state ownership guarantees.
+
+Contract composition has led to multiple multi-million dollar bugs in Ethereum
+
+* https://www.parity.io/security-alert-2/
+
+## Scripts
+
+The purpose of scripts is to execute logic between instructions.
+
+For example, assignment of tokens to the winner based on an output of a game can
+be defined as a script.
+
+### `Loader::CreateScript`
+
+`Loader::CreateScript` designates that the loaded executable bytecode is a
+script. The difference between scripts and programs is that script execution
+yields to external process instructions.
+
+### Script Execution
+
+The general idea to script execution is that the client knows a head of time
+what branches the script will take, and therefore can encode all the expected
+instructions the script will generate into the transaction.
+
+Transactions that include instructions that invoke scripts encode all the
+instructions that the script will make during its execution in the message
+instruction vector.  The additional instructions appear before the script, in the
+exact same order and the same parameters as the script would make during
+execution.
+
+During the script execution, calls to instructions "yield", and the next
+instruction to be processed in the instruction vector is invoked.  The
+instruction that is called must match the one that the script is expecting to
+call through its execution path.  If the instruction doesn't match, the program
```

Comment:
clunky

Reply:
I added an explanation why I think this is a requirement.  The basic idea is that a dynamic path on chain is pretty scary.  Users should authorize a specific path with each transaction explicitly.